File: P_6/Figuras.java
import java.util.Scanner;

abstract class FiguraGeometrica {
    public abstract double calcularArea();
    public abstract double calcularPerimetro();
}

class Triangulo extends FiguraGeometrica {
    private double base;
    private double altura;

    public Triangulo(double base, double altura) {
        this.base = base;
        this.altura = altura;
    }

    public double calcularArea() {
        return (base * altura) / 2;
    }

    public double calcularPerimetro() {
        return 3 * base;
    }
}

class Circulo extends FiguraGeometrica {
    private double radio;

    public Circulo(double radio) {
        this.radio = radio;
    }

    public double calcularArea() {
        return Math.PI * Math.pow(radio, 2);
    }

    public double calcularPerimetro() {
        return 2 * Math.PI * radio;
    }
}

class Rectangulo extends FiguraGeometrica {
    private double ladoA;
    private double ladoB;

    public Rectangulo(double ladoA, double ladoB) {
        this.ladoA = ladoA;
        this.ladoB = ladoB;
    }

    public double calcularArea() {
        return ladoA * ladoB;
    }

    public double calcularPerimetro() {
        return 2 * (ladoA + ladoB);
    }
}

class Hexagono extends FiguraGeometrica {
    private double lado;

    public Hexagono(double lado) {
        this.lado = lado;
    }

    public double calcularArea() {
        return (Math.sqrt((lado*lado)-((lado/2)*(lado/2)))*(lado)*6)/2;
    }

    public double calcularPerimetro() {
        return 6 * lado;
    }
}

public class Figuras {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("----- Menú de Opciones -----");
            System.out.println("1. Calcular área y perímetro de un triángulo");
            System.out.println("2. Calcular área y perímetro de un círculo");
            System.out.println("3. Calcular área y perímetro de un rectángulo");
            System.out.println("4. Calcular área y perímetro de un hexágono");
            System.out.println("5. Salir");
            System.out.print("Seleccione una opción: ");

            String opcion = scanner.nextLine();

            if (opcion.equals("1")) {
                System.out.print("Ingrese la base del triángulo: ");
                double base = scanner.nextDouble();
                System.out.print("Ingrese la altura del triángulo: ");
                double altura = scanner.nextDouble();

                FiguraGeometrica triangulo = new Triangulo(base, altura);
                System.out.println("Área: " + triangulo.calcularArea());
                System.out.println("Perímetro: " + triangulo.calcularPerimetro());

            } else if (opcion.equals("2")) {
                System.out.print("Ingrese el radio del círculo: ");
                double radio = scanner.nextDouble();

                FiguraGeometrica circulo = new Circulo(radio);
                System.out.println("Área: " + circulo.calcularArea());
                System.out.println("Perímetro: " + circulo.calcularPerimetro());

            } else if (opcion.equals("3")) {
                System.out.print("Ingrese el lado A del rectángulo: ");
                double ladoA = scanner.nextDouble();
                System.out.print("Ingrese el lado B del rectángulo: ");
                double ladoB = scanner.nextDouble();
                
                FiguraGeometrica rectangulo = new Rectangulo(ladoA, ladoB);
                System.out.println("Área: " + rectangulo.calcularArea());
                System.out.println("Perímetro: " + rectangulo.calcularPerimetro());

            } else if (opcion.equals("4")) {
                System.out.print("Ingrese el lado del hexágono: ");
                double lado = scanner.nextDouble();
                
                FiguraGeometrica hexagono = new Hexagono(lado);
                System.out.println("Área: " + hexagono.calcularArea());
                System.out.println("Perímetro: " + hexagono.calcularPerimetro());

            } else if (opcion.equals("5")) {
                break;

            } else {
                System.out.println("Opción inválida. Por favor, seleccione una opción válida.");
                
            }

            System.out.println();
            scanner.nextLine(); // Consumir el salto de línea después de leer un valor numérico
        }

        scanner.close();
    }
}

File: P_6/figuras.py
from abc import ABC, abstractmethod
import math

class FiguraGeometrica(ABC):
    @abstractmethod
    def calcular_area(self):
        pass

    @abstractmethod
    def calcular_perimetro(self):
        pass

class Triangulo(FiguraGeometrica):
    def __init__(self, base, altura):
        self.base = base
        self.altura = altura

    def calcular_area(self):
        return (self.base * self.altura) / 2

    def calcular_perimetro(self):
        return 3 * self.base

class Circulo(FiguraGeometrica):
    def __init__(self, radio):
        self.radio = radio

    def calcular_area(self):
        return math.pi * (self.radio ** 2)

    def calcular_perimetro(self):
        return 2 * math.pi * self.radio

class Rectangulo(FiguraGeometrica):
    def __init__(self, lado_a, lado_b):
        self.lado_a = lado_a
        self.lado_b = lado_b

    def calcular_area(self):
        return self.lado_a * self.lado_b

    def calcular_perimetro(self):
        return 2 * (self.lado_a + self.lado_b)
    
class Hexagono(FiguraGeometrica):
    def __init__(self, lado):
        self.lado = lado

    def calcular_area(self):
        return (math.sqrt((self.lado*self.lado)-((self.lado/2)*(self.lado/2))))*(self.lado*6)/2

    def calcular_perimetro(self):
        return 6 * self.lado

# Función para mostrar el menú y realizar cálculos
def menu():
    while True:
        print("----- Menú de Opciones -----")
        print("1. Calcular área y perímetro de un triángulo")
        print("2. Calcular área y perímetro de un círculo")
        print("3. Calcular área y perímetro de un rectángulo")
        print("4. Calcular área y perímetro de un hexagono")
        print("5. Salir")
        opcion = input("Seleccione una opción: ")

        if opcion == "1":
            base = float(input("Ingrese la base del triángulo: "))
            altura = float(input("Ingrese la altura del triángulo: "))
            triangulo = Triangulo(base, altura)
            print("Área:", triangulo.calcular_area())
            print("Perímetro:", triangulo.calcular_perimetro())

        elif opcion == "2":
            radio = float(input("Ingrese el radio del círculo: "))
            circulo = Circulo(radio)
            print("Área:", circulo.calcular_area())
            print("Perímetro:", circulo.calcular_perimetro())

        elif opcion == "3":
            lado_a = float(input("Ingrese el lado A del rectángulo: "))
            lado_b = float(input("Ingrese el lado B del rectángulo: "))
            rectangulo = Rectangulo(lado_a, lado_b)
            print("Área:", rectangulo.calcular_area())
            print("Perímetro:", rectangulo.calcular_perimetro())

        
        elif opcion == "4":
            lado = float(input("Ingrese el lado del hexagono: "))
            hexagono = Hexagono(lado)
            print("Área:", hexagono.calcular_area())
            print("Perímetro:", hexagono.calcular_perimetro())

        elif opcion == "5":
            break

        else:
            print("Opción inválida. Por favor, seleccione una opción válida.")

        print()

# Ejecutar el menú
menu()